src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -183,7 +183,7 @@ private Thread startDirectoryWatcher(Path mapsDirectory) {
       WatchService watcher = mapsDirectory.getFileSystem().newWatchService();
       customMapsDirectory.register(watcher, ENTRY_DELETE);
 
-      try {
+      try (watcher) {
```

Comment:
You can declare the variable in the try-with-resources block. 
`WatchService watcher = mapsDirectory.getFileSystem().newWatchService();` goes into the try()
  `    customMapsDirectory.register(watcher, ENTRY_DELETE);` becomes part of the try-block

Reply:
oh yeah u sure, no idea why i did it like that